evalml/tests/data_checks_tests/test_datetime_format_data_check.py

```diff
@@ -0,0 +1,130 @@
+import pandas as pd
+import pytest
+
+from evalml.data_checks import (
+    DataCheckError,
+    DataCheckMessageCode,
+    DateTimeFormatDataCheck,
+)
+
+datetime_format_check_name = DateTimeFormatDataCheck.name
+
+
+@pytest.mark.parametrize("input_type", ["pd", "ww"])
+@pytest.mark.parametrize(
+    "uneven,type_errors", [(True, False), (False, True), (False, False)]
+)
+@pytest.mark.parametrize("datetime_loc", ["datetime_feature", "X_index", "y_index"])
+def test_datetime_format_data_check_typeerror_uneven_intervals(
+    uneven, input_type, type_errors, datetime_loc
+):
+    X, y = pd.DataFrame({"features": range(30)}), pd.Series(range(30))
+
+    if type_errors:
+        dates = range(30)
+    else:
+        dates = pd.date_range("2021-01-01", periods=30)
+
+    if uneven:
+        dates = list(dates)
+        dates[-1] = "2021-01-31"
+        dates = pd.DatetimeIndex(dates)
+
+    datetime_column = "index"
+    if datetime_loc == "datetime_feature":
+        X[datetime_loc] = dates
+        datetime_column = datetime_loc
+    elif datetime_loc == "X_index":
+        X.index = dates
+    else:
+        y.index = dates
+
+    if input_type == "ww":
+        X.ww.init()
+        y.ww.init()
+
+    datetime_format_check = DateTimeFormatDataCheck(datetime_column=datetime_column)
+
+    if type_errors:
+        if datetime_loc == "datetime_feature":
+            with pytest.raises(
+                TypeError,
+                match="That column does not contain datetime",
+            ):
+                datetime_format_check.validate(X, y)
+        else:
+            with pytest.raises(
+                TypeError,
+                match="Either X or y has to have datetime information in its index.",
+            ):
+                datetime_format_check.validate(X, y)
+    else:
+        if not uneven:
+            assert datetime_format_check.validate(X, y) == {
+                "warnings": [],
+                "errors": [],
+                "actions": [],
+            }
+        else:
+            message = (
```

Comment:
Same nitpick, maybe rename this to col_name or something? (Since we're using it as a formatted parameter in a parameter called `message` 😂 )